prov/util/src/util_cntr.c

```diff
@@ -157,11 +157,8 @@ static int ofi_cntr_wait(struct fid_cntr *cntr_fid, uint64_t threshold, int time
 		if (errcnt != ofi_atomic_get64(&cntr->err))
 			return -FI_EAVAIL;
 
-		if (timeout >= 0) {
-			timeout -= (int) (fi_gettime_ms() - start);
-			if (timeout <= 0)
-				return -FI_ETIMEDOUT;
-		}
+		if (ofi_adjust_timeout(endtime, &timeout))
```

Comment:
We should probably not be modifying the given argument to avoid surprising the caller. We can copy it to a local variable and modify it instead.

Reply:
Shouldn't the timeout arg be const?

Reply:
The user's argument isn't being modified.  Timeout is passed to the call by value.  We're only updating the stack copy here.

Reply:
Ah, that's right. :)